pkgs/by-name/no/noriskclient-launcher/package.nix

```diff
@@ -0,0 +1,79 @@
+{
+  lib,
+  fetchFromGitHub,
+
+  fetchNpmDeps,
+  npmHooks,
+  nodejs,
+
+  rustPlatform,
+  cargo-tauri,
+
+  pkg-config,
+  wrapGAppsHook3,
+
+  openssl,
+  libsoup,
+  webkitgtk_4_0,
+  glib-networking,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "noriskclient-launcher";
+  version = "0.5.10";
+
+  src = fetchFromGitHub {
+    owner = "NoRiskClient";
+    repo = "noriskclient-launcher";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-WaDAP7Oy28QY4wKVXYW82wN1wCT3QhkBXeUoEiSIfME=";
+  };
+
+  npmDeps = fetchNpmDeps {
+    name = "${pname}-${version}-npm-deps";
+    inherit src;
+    forceGitDeps = true;
+    hash = "sha256-Rw8iyvnG0tK5b/tJ5fg5aS7HIvFuDHFRuWSpzAEq8uA=";
+  };
+
+  forceGitDeps = true;
+
+  makeCacheWritable = true;
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "fix-path-env-0.0.0" = "sha256-kSpWO2qMotpsYKJokqUWCUzGGmNOazaREDLjke4/CtE=";
+      "tauri-plugin-fs-watch-0.0.0" = "sha256-4Y0Vbd14QWqI3NFcgvfMrXVvGvuA1spXibCXpjHdR9s=";
+    };
+  };
+
+  cargoRoot = "src-tauri";
+
+  buildAndTestSubdir = cargoRoot;
+
+  nativeBuildInputs = [
+    npmHooks.npmConfigHook
+    nodejs
+
+    cargo-tauri.hook
+
+    pkg-config
+    wrapGAppsHook3
+  ];
+
+  buildInputs = [
+    openssl
+    libsoup
+    webkitgtk_4_0
+    glib-networking
+  ];
+
+  meta = {
+    description = "Minecraft client application made using svelte + tauri";
+    homepage = "https://norisk.gg";
+    license = lib.licenses.gpl3Only;
```

Comment:
While the launcher is open source and code implies support of multiple mod loaders, I believe it's going to *always* inject some kind of proprietary mod into modpacks, shouldn't this be somehow added in meta?

Reply:
We don't currently have a really good way to imply this. But the argument is not in the mods, but in Minecraft itself - browsers and IDEs are also allowed to install closed source plug-ins, and we have not blocked the download channels provided in their apps to obtain them. The Minecraft launcher itself is incomplete. It requires the program ontology to form a complete program. But from another perspective, you can also think that it is complete in terms of its functionality - just launch Minecraft. So I don't think this question can be given a very good answer here.